lib/core/src/sdk.rs

```diff
@@ -3138,21 +3138,34 @@ impl LiquidSdk {
             .persister
             .get_is_first_sync_complete()?
             .unwrap_or(false);
+
+        let sync_payments_with_chain_data = async {
+            match self.sync_payments_with_chain_data(partial_sync).await {
+                Ok(_) => {
+                    if is_first_sync {
+                        self.persister.set_is_first_sync_complete(true)?;
+                    }
+                    Result::<()>::Ok(())
+                }
+                Err(err) => anyhow::bail!("Could not sync payments with chain data: {err:?}"),
+            }
+        };
```

Comment:
is this async code necessary? My understanding is the only change we need here is to not set the first sync as completed in case of error.